### PR TITLE
Fixed issues with encoder and decoder [variable length]

### PR DESCRIPTION
**Pull request by premsantosh**

### CODE REVIEW

File: lang/py/src/avro/io.py [hunk 1]
```diff
@@ -407,53 +394,60 @@ def write_decimal_bytes(self, datum, scale):
...
+      unscaled_datum *= 10
+      unscaled_datum += digit
+
+    bits_req = unscaled_datum.bit_length()
+    if sign:
+      bits_req += 1
+      unscaled_datum = (1 << bits_req) - unscaled_datum
+
+    bytes_req = bits_req // 8
+    bytes_req += 1 if (bytes_req << 3) < bits_req else 0
```
Comment (clin7) [inline lang/py/src/avro/io.py:441]: The bytes_req should be the given fixed size. It is possible the number of padding bits are more than 1 byte.

File: lang/py/src/avro/schema.py [hunk 2]
```diff
@@ -857,6 +857,8 @@ def make_avsc_object(json_data, names=None):
         if logical_type == 'decimal':
           precision = json_data.get('precision')
           scale = 0 if json_data.get('scale') is None else json_data.get('scale')
+          other_props['name'] = name
+          other_props['namespace'] = namespace
```
Comment (clin7) [inline lang/py/src/avro/schema.py]: The name and namespace are the required fields of fixed schema. They're not the other_props.

File: lang/py/src/avro/io.py (hunk 1 above)
Comment (premsantosh) [inline lang/py/src/avro/io.py:441]: I have that on #447. This just checks if the fixed size can fit the decimal or not.

File: lang/py/src/avro/schema.py (hunk 2 above)
Comment (premsantosh) [inline lang/py/src/avro/schema.py]: You're right. Will change this.

File: lang/py/src/avro/io.py (hunk 1 above)
Comment (premsantosh) [inline lang/py/src/avro/io.py:441]: On the other hand, the size shouldn't change. I need to fix this.

**Comment by clin7** [comment]
Looks good to me, Ship it.

**Pull request merged**